src/domain/core/src/entities/dataset.rs

```diff
@@ -148,3 +165,59 @@ pub enum CommitError {
         InternalError,
     ),
 }
+
+/////////////////////////////////////////////////////////////////////////////////////////
+// Commit helpers
+/////////////////////////////////////////////////////////////////////////////////////////
+
+/// Replicates [AddData] event prior to hasing of data and checkpoint
+pub struct AddDataParams {
+    pub input_checkpoint: Option<Multihash>,
+    pub output_data: Option<OffsetInterval>,
+    pub output_watermark: Option<DateTime<Utc>>,
+    pub source_state: Option<SourceState>,
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+/// Replicates [ExecuteQuery] event prior to hasing of data and checkpoint
+pub struct ExecuteQueryParams {
+    pub input_slices: Vec<InputSlice>,
+    pub input_checkpoint: Option<Multihash>,
+    pub output_data: Option<OffsetInterval>,
+    pub output_watermark: Option<DateTime<Utc>>,
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+/// Represents a file on local FS that should either be consumed using
+/// [OwnedFile::into_inner()] or will be cleaned up on drop. Is typically used
+/// to commit data and checkopints generated by the engines into a dataset.
+#[derive(Debug)]
+pub struct OwnedFile {
+    path: Option<PathBuf>,
+}
+
+impl OwnedFile {
+    pub fn new(path: impl Into<PathBuf>) -> Self {
+        let path = path.into();
+        assert!(path.exists());
+        Self { path: Some(path) }
+    }
+
```

Comment:
I guess this should be isolated in `domain/core/src/utils` rather then staying in the `dataset.rs`

Reply:
done!